### PR TITLE
 mldonkey: 3.1.6 -> 3.1.7-2 

### PR DESCRIPTION
**Pull request by sternenseemann**
###### Motivation for this change

Be able to use a more recent OCaml version for building.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 120776` at c5305c47 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>mldonkey</li>
  </ul>
</details>
<details>
  <summary>7 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `bzip2`.
Near pkgs/applications/networking/p2p/mldonkey/default.nix:1:47:
```
  |
1 | { lib, stdenv, fetchurl, ocamlPackages, zlib, bzip2, ncurses, file, gd, libpng, libjpeg }:
  |                                               ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/networking/p2p/mldonkey/default.nix:25:3:
```
   |
25 |   meta = {
   |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `libjpeg`.
Near pkgs/applications/networking/p2p/mldonkey/default.nix:1:81:
```
  |
1 | { lib, stdenv, fetchurl, ocamlPackages, zlib, bzip2, ncurses, file, gd, libpng, libjpeg }:
  |                                                                                 ^
```
</li>
    <li>warning: unused-argument

Unused argument: `ncurses`.
Near pkgs/applications/networking/p2p/mldonkey/default.nix:1:54:
```
  |
1 | { lib, stdenv, fetchurl, ocamlPackages, zlib, bzip2, ncurses, file, gd, libpng, libjpeg }:
  |                                                      ^
```
</li>
    <li>warning: unused-argument

Unused argument: `gd`.
Near pkgs/applications/networking/p2p/mldonkey/default.nix:1:69:
```
  |
1 | { lib, stdenv, fetchurl, ocamlPackages, zlib, bzip2, ncurses, file, gd, libpng, libjpeg }:
  |                                                                     ^
```
</li>
    <li>warning: unused-argument

Unused argument: `libpng`.
Near pkgs/applications/networking/p2p/mldonkey/default.nix:1:73:
```
  |
1 | { lib, stdenv, fetchurl, ocamlPackages, zlib, bzip2, ncurses, file, gd, libpng, libjpeg }:
  |                                                                         ^
```
</li>
    <li>warning: unused-argument

Unused argument: `file`.
Near pkgs/applications/networking/p2p/mldonkey/default.nix:1:63:
```
  |
1 | { lib, stdenv, fetchurl, ocamlPackages, zlib, bzip2, ncurses, file, gd, libpng, libjpeg }:
  |                                                               ^
```
</li>
  </ul>
</details>

File: pkgs/applications/networking/p2p/mldonkey/default.nix [hunk 1]
```diff
@@ -1,37 +1,31 @@
...
-stdenv.mkDerivation ({
-  name = "mldonkey-3.1.6";
+stdenv.mkDerivation rec {
+  pname = "mldonkey";
+  version = "3.1.7-2";
 
   src = fetchurl {
-    url = "https://github.com/ygrek/mldonkey/releases/download/release-3-1-6/mldonkey-3.1.6.tar.bz2";
-    sha256 = "0g84islkj72ymp0zzppcj9n4r21h0vlghnq87hv2wg580mybadhv";
+    url = "https://ygrek.org/p/release/mldonkey/mldonkey-${version}.tar.bz2";
```
Comment (karolchmist) [inline pkgs/applications/networking/p2p/mldonkey/default.nix:8]: Any reason why not use the github link? It is the same as far as I can tell...
https://github.com/ygrek/mldonkey/releases/download/release-3-1-7-2/mldonkey-3.1.7-2.tar.bz2

**Comment by karolchmist** [comment]
I built it and run successfully, looks good!

File: pkgs/applications/networking/p2p/mldonkey/default.nix (hunk 1 above)
Comment (sternenseemann) [reply pkgs/applications/networking/p2p/mldonkey/default.nix:8]: That is the URL listed in opam-repository which I used…

**Pull request merged**